helm/prometheus-rules/templates/alerting-rules/capo.rules.yml

```diff
@@ -0,0 +1,46 @@
+{{- if or (eq .Values.managementCluster.provider.kind "openstack") }}
+kind: PrometheusRule
+metadata:
+  labels: {{- include "labels.common" . | nindent 4}}
+  name: capi.rules
+  namespace: {{.Values.namespace}}
+spec:
+  groups:
+    - name: coredns
+      rules:
+        - alert: OpenStackMachineUnexpectedInstanceState
+          expr: capi_openstackmachine_status_instance_state{state!="ACTIVE"} > 0
+          for: 1h
+          labels:
+            area: kaas
+            cancel_if_outside_working_hours: {{include "workingHoursOnly" .}}
+            severity: notify
+            team: {{include "providerTeam" .}}
+            topic: managementcluster
+          annotations:
+            description: "{{`OpenStackMachine {{$labels.namespace}}/{{$labels.name}} instance state is {{$labels.state}}.`}}"
+
+        - alert: OpenStackMachineFailure
+          expr: capi_openstackmachine_status_failure{failure!="<nil>"}
+          for: 30m
```

Comment:
MachineHealhCheck.spec looks like this currently:

```
spec:
  clusterName: gawel
  maxUnhealthy: 40%
  nodeStartupTimeout: 20m0s
  selector:
    matchLabels:
      cluster.x-k8s.io/cluster-name: gawel
  unhealthyConditions:
  - status: Unknown
    timeout: 10m0s
    type: Ready
  - status: "False"
    timeout: 10m0s
    type: Ready
```

I understand it may take 30m till it's triggered. Shouldn't we increase the duration here?

Reply:
`MachineHealhChecks` doesn't solve this as they are only take care of conditions. As conditions in CAPO are currently in implementing phase (:crossed_fingers: they will land in the next release) this is alert is only there to detect the transient errors. A possible opsrecipe could be _"patch the status of this `osm`"_

Reply:
Speaking of that, I guess we should add opsrecipes eveywhere?